frontend/src/views/AdminPages/BranchesPage/BranchesPage.js

```diff
@@ -0,0 +1,143 @@
+import Button from "components/CustomButtons/Button.js";
+import SneakingComponent from "../SneakingComponent/SneakingComponent"
+import styles from "assets/jss/material-kit-react/views/loginPage.js";
+import bgimage from "assets/img/bg7.jpg";
+import Danger from "components/Typography/Danger";
+import BranchComponent from "./BranchComponent/BranchComponent";
+import BranchEditComponent from "./BranchEditComponent/BranchEditComponent";
+import Loader from "components/Loader/Loader";
+
+const useStyles = makeStyles(styles);
+
+export default function BranchesPage(props) {
+
+    const classes = useStyles();
+    const { ...rest } = props;
+
+    const [modalVisible, setModalVisible] = useState(false);
+    const [branch, setBranch] = useState();
+
+    const userSignin = useSelector(state => state.userSignin);
+    const { userInfo } = userSignin
+    const branchesList = useSelector(state => state.branchesList);
+    const { loading, branches, error } = branchesList;
+    const branchSave = useSelector(state => state.branchSave);
+    const { loading: loadingSave, success: successSave, error: errorSave } = branchSave;
+    const branchDelete = useSelector(state => state.branchDelete);
+    const { success: successDelete } = branchDelete;
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        if (successSave) {
+            setModalVisible(false);
+        }
+        dispatch(listBranches());
+        return () => {
+            //
+        };
+    }, [successSave, successDelete]);
+
+    const closeModal = () => {
+        setModalVisible(false);
+    }
+    const openModal = (branch) => {
+        setBranch(branch);
+        setModalVisible(true);
+    }
+
+    const deleteHandler = (branch) => {
+        dispatch(deleteBranch(branch._id));
+    }
+
+    return (
+        <div>
+            {userInfo && userInfo.isAdmin ?
+                <>
+                    <Header
+                        absolute
+                        color="transparent"
+                        brand="Material Kit React"
+                        rightLinks={<HeaderLinks />}
+                        {...rest}
+                    />
+                    <div
+                        className={classes.pageHeader}
+                        style={{
+                            backgroundImage: "url(" + bgimage + ")",
+                            backgroundSize: "cover",
+                            backgroundPosition: "top center",
+                        }}
+                    >
+                        <div className={classes.container}>
+                            <Card>
+                                <form className={classes.form}>
+                                    <GridContainer justify="center">
+                                        <GridItem xs={12} sm={12} md={11}>
+                                            {/* did this to keep the centered styling as in login page, while allowing the body to go wide */}
```

Comment:
Even though the code is self explanatory, I might attempt to restructure the layout in the future and ruin what I set. That's why I initially put the comment here. Do you think that even these comments should be removed?